### ISSUE TITLE
Fitting before predicting?

### ISSUE THREAD

**Issue by mherde** [scikit-activeml/scikit-activeml#40]
Hey guys,

I face the issue when an estimator should be able to make predictions. In the current version, an estimator is able to make predictions without calling `fit` as long as `classes` has been given as parameter. This is not compatible to Scikit-learn. If I would remove this functionality, the estimator can be still fitted on data with no labels (i.e., `y` contains only missing labels) and can make predictions afterwards. What do you think?

**Comment by dakot** [comment]
I think, it is fine that the classifier is only able to predict when it was fitted.

**Comment by mherde** [comment]
Okay, I think so, too. The only difference would be that `PWC(classes=[0,1]).predict(X)` is not possible anymore but you have to use: `PWC(classes=[0,1]).fit(X, y).predict(X)` where `y` may exclusively consist of missing labels.

**Comment by tpham93** [comment]
Didn't we say, that we would have a wrapper for sklearn classifiers either way? If yes, it should be possible to wrap the prediction functions to output a random class, if not the base model is not fitted yet.
However, on second thought, that would render the check_is_fitted useless, as model-specific parameters may still not be initialized correctly.

**Comment by dakot** [comment]
I think that there is no use of getting predictions without having fitted a classifier anyway. at the beginning of each AL cycle, one would fit the classifier nevertheless.

**Comment by mherde** [comment]
Yes, currently, all estimators (including the wrapper) are making random predictions if not fitted. But, as you said, it is somehow contradicting to the intention of check_is_fitted.
